xtrack/beam_elements/apertures.py

```diff
@@ -57,12 +86,18 @@ def to_dict(self):
         return dct
 
     def __init__(self, a=None, b=None, a_squ=None, b_squ=None, **kwargs):
+        """An elliptical aperture
+        
+        Args:
+            a (float): Horizontal semi-axis of ellipse in m
+            b (float): Vertical semi-axis of ellipse in m
+        """
 
         if a is None and a_squ is None:
-            a = 1.
+            raise ValueError("Horizontal semi-axis of ellipse not specified.")
```

Comment:
Why don't we use "UNLIMITED" also here?

Reply:
In my opinion, it is pointless to have an elliptic aperture with one dimension unlimited, since there is nothing elliptical about this aperture anymore, it's just a slit. One should use a LimitRect instead, which is faster in tracking, clearer in code, and - in contrast to the "ellipse" - numerically stable[^1]. If one of the semi-axes are not given, it's likely the user forgot it.

What we could do, instead of raising an Exception, is to default to a circular aperture (i.e. a=b), if you want?

[^1]: The ellipse with one very large semi-axis is numerically unstable, since in the code `double const temp = x*x*b_squ + y*y*a_squ;`, the small float being added to the large float will be lost due to insufficient precision.

Reply:
I will merge it with UNLIMITED for now (I reduced UNLIMITED to 1e10). We can change it later